### ISSUE TITLE
Start numeral of a list, specified with `[@3 ]` etc is not honored

### ISSUE THREAD

**Issue by kjambunathan** [kjambunathan/org-mode-ox-odt#62]
## Start numeral of a list, specified with `[@3 ]` etc is not honored

According to [info "(org) Plain lists")](https://www.gnu.org/software/emacs/manual/html_node/org/Plain-lists.html#Plain-lists),

> Ordered list items start with a numeral followed by either a period or a right parenthesis2, such as ‘1.’ or ‘1)’3. If you want a list to start with a different value (e.g., 20), start the text of the item with [@20]4. Those constructs can be used in any item of the list in order to enforce a particular numbering. 

Export the following snippet to ODT

```
        Some text
    
        1. First item
        2. Second item
    
        Some intervening text
    
        3. [@3] Third item
        4. Fourth item

```

The `Third item` should have a number of 3.  But currently, it starts with 1.

**Comment by revrari** [comment]
I applied the fix and the file is now exported correctly. One minor inconvenience is that I am now prompted with the following error message which was not there before the fix. 

![corrupt-file](https://user-images.githubusercontent.com/44517802/69474794-dcacf280-0dcd-11ea-81cd-65ee43392636.png)

**Comment by kjambunathan** [comment]
>  One minor inconvenience is that I am now prompted with the following error message which was not there before the fix.

There is a note about handling corrupt files in the manual. See [(info "(org) Validating OpenDocument XML")](https://www.gnu.org/software/emacs/manual/html_node/org/Validating-OpenDocument-XML.html).  It goes something like 

> Sometimes ODT format files may not open due to .odt file corruption. To verify if the .odt file is corrupt, validate it against the OpenDocument RELAX NG Compact Syntax—RNC—schema. But first the .odt files have to be decompressed using ‘zip’. Note that .odt files are ‘zip’ archives: see File Archives. The contents of .odt files are in .xml. For general help with validation—and schema-sensitive editing—of XML files: see Introduction. 

> Customize org-odt-schema-dir to point to a directory with OpenDocument .rnc files and the needed schema-locating rules. The ODT export back-end takes care of updating the rng-schema-locating-files. 

The paragraphs above may be too crpytic or terse for someone doing it for the first time.

This is what you need to do:

1. Export your `test.org`.
2. Your `*Messages*` buffer will tell you what the `odt` backend is doing.  Specifically it will tell you where it is picking the `styles` and `schema` files from.  Ensure that they are from your ELPA path i.e., `~/.emacs.d/elpa/org-mode-ox-odt/...`.  If the path is not from ELPA path, then stop.  You need to check your init file and fix your `load-path` so that the `ox-odt` that gets used for export is actually from ELPA installation.  The commands  `M-x locate-library RET ox-odt RET` and `M-x list-load-path-shadows RET` will help you identify `load-path` related issues.  Apart from what I say here, you can also consult the note https://github.com/kjambunathan/org-mode-ox-odt/issues/48#issuecomment-441103145 and subsequent notes for other useful hints.
3. C-x C-f test.odt
2. If the `odt` file opens in `doc-view-mode` (just like a `pdf` file), you need to switch the bufffer to `archive-mode`.  You can switch mode by using `C-c C-c` in the `doc-view-mode`.
3.  In the resulting buffer, locate `content.xml` and press `RET`.
4. Open the `content.xml` file, and  pay attention to the `mode-line`.  See if it says `valid` or `invalid`.  The schema files reported in step (2) are used for validation.  In your case, it will say `invalid`.  Then press `C-c C-n` to navigate to the first error.   Once you have identified the error, you will know what to do.

Here is a screenshot that summarizes all the above points.

![Screenshot from 2019-11-23 18-47-11](https://user-images.githubusercontent.com/19869864/69479977-7fdf2580-0e29-11ea-851e-d97c701ba78a.png)

**Comment by kjambunathan** [comment]
> Open the content.xml file, and pay attention to the mode-line. See if it says valid or invalid.

Usually it is the `content.xml` file that is corrupt.  Occasionally `styles.xml` or `manifest.xml` may also get corrupt.  So, you may want to run the _validation_ checks on not just `content.xml` but also all the other `xml` files in the ODT file.

**Comment by revrari** [comment]
I checked the ELPA path as per instructions above and everything is correct. I also checked the .xml files and they are reported as valid. I am enclosing the following information for your interest. M-x list-load-path-shadows produced:

/home/red/.emacs.d/elpa/ox-odt-9.2.6.260/ox-odt hides /home/red/.emacs.d/elpa/org-20191118/ox-odt
/home/red/.emacs.d/elpa/org-20191118/org-clock hides /home/red/emacs/lisp/org/org-clock
/home/red/.emacs.d/elpa/org-20191118/ob-C hides /home/red/emacs/lisp/org/ob-C
/home/red/.emacs.d/elpa/org-20191118/ob-shell hides /home/red/emacs/lisp/org/ob-shell
/home/red/.emacs.d/elpa/org-20191118/ob-sqlite hides /home/red/emacs/lisp/org/ob-sqlite
/home/red/.emacs.d/elpa/org-20191118/ob-lilypond hides /home/red/emacs/lisp/org/ob-lilypond
/home/red/.emacs.d/elpa/org-20191118/ob-table hides /home/red/emacs/lisp/org/ob-table
/home/red/.emacs.d/elpa/org-20191118/ob-keys hides /home/red/emacs/lisp/org/ob-keys
/home/red/.emacs.d/elpa/org-20191118/org-indent hides /home/red/emacs/lisp/org/org-indent
/home/red/.emacs.d/elpa/org-20191118/org-capture hides /home/red/emacs/lisp/org/org-capture
/home/red/.emacs.d/elpa/org-20191118/ob-matlab hides /home/red/emacs/lisp/org/ob-matlab
/home/red/.emacs.d/elpa/org-20191118/ob-eval hides /home/red/emacs/lisp/org/ob-eval
/home/red/.emacs.d/elpa/org-20191118/org-eshell hides /home/red/emacs/lisp/org/org-eshell
/home/red/.emacs.d/elpa/org-20191118/org-lint hides /home/red/emacs/lisp/org/org-lint
/home/red/.emacs.d/elpa/org-20191118/ob-coq hides /home/red/emacs/lisp/org/ob-coq
/home/red/.emacs.d/elpa/org-20191118/org-gnus hides /home/red/emacs/lisp/org/org-gnus
/home/red/.emacs.d/elpa/org-20191118/ob-sql hides /home/red/emacs/lisp/org/ob-sql
/home/red/.emacs.d/elpa/org-20191118/ob-ruby hides /home/red/emacs/lisp/org/ob-ruby
/home/red/.emacs.d/elpa/org-20191118/org-bibtex hides /home/red/emacs/lisp/org/org-bibtex
/home/red/.emacs.d/elpa/org-20191118/org-rmail hides /home/red/emacs/lisp/org/org-rmail
/home/red/.emacs.d/elpa/org-20191118/ob-comint hides /home/red/emacs/lisp/org/ob-comint
/home/red/.emacs.d/elpa/org-20191118/org-feed hides /home/red/emacs/lisp/org/org-feed
/home/red/.emacs.d/elpa/org-20191118/ob-abc hides /home/red/emacs/lisp/org/ob-abc
/home/red/.emacs.d/elpa/org-20191118/ob-makefile hides /home/red/emacs/lisp/org/ob-makefile
/home/red/.emacs.d/elpa/org-20191118/org hides /home/red/emacs/lisp/org/org
/home/red/.emacs.d/elpa/org-20191118/org-compat hides /home/red/emacs/lisp/org/org-compat
/home/red/.emacs.d/elpa/org-20191118/ob-lob hides /home/red/emacs/lisp/org/ob-lob
/home/red/.emacs.d/elpa/org-20191118/ox-icalendar hides /home/red/emacs/lisp/org/ox-icalendar
/home/red/.emacs.d/elpa/org-20191118/ob-groovy hides /home/red/emacs/lisp/org/ob-groovy
/home/red/.emacs.d/elpa/org-20191118/ob-stan hides /home/red/emacs/lisp/org/ob-stan
/home/red/.emacs.d/elpa/org-20191118/org-mouse hides /home/red/emacs/lisp/org/org-mouse
/home/red/.emacs.d/elpa/org-20191118/ob-octave hides /home/red/emacs/lisp/org/ob-octave
/home/red/.emacs.d/elpa/org-20191118/ob-calc hides /home/red/emacs/lisp/org/ob-calc
/home/red/.emacs.d/elpa/org-20191118/ob-python hides /home/red/emacs/lisp/org/ob-python
/home/red/.emacs.d/elpa/org-20191118/ob-emacs-lisp hides /home/red/emacs/lisp/org/ob-emacs-lisp
/home/red/.emacs.d/elpa/org-20191118/ob-haskell hides /home/red/emacs/lisp/org/ob-haskell
/home/red/.emacs.d/elpa/org-20191118/ob-lisp hides /home/red/emacs/lisp/org/ob-lisp
/home/red/.emacs.d/elpa/org-20191118/org-irc hides /home/red/emacs/lisp/org/org-irc
/home/red/.emacs.d/elpa/org-20191118/ob-css hides /home/red/emacs/lisp/org/ob-css
/home/red/.emacs.d/elpa/org-20191118/org-habit hides /home/red/emacs/lisp/org/org-habit
/home/red/.emacs.d/elpa/org-20191118/ob-vala hides /home/red/emacs/lisp/org/ob-vala
/home/red/.emacs.d/elpa/org-20191118/ob-picolisp hides /home/red/emacs/lisp/org/ob-picolisp
/home/red/.emacs.d/elpa/org-20191118/org-mhe hides /home/red/emacs/lisp/org/org-mhe
/home/red/.emacs.d/elpa/ox-odt-9.2.6.260/ox-odt hides /home/red/emacs/lisp/org/ox-odt
/home/red/.emacs.d/elpa/org-20191118/org-table hides /home/red/emacs/lisp/org/org-table
/home/red/.emacs.d/elpa/org-20191118/org-footnote hides /home/red/emacs/lisp/org/org-footnote
/home/red/.emacs.d/elpa/org-20191118/org-entities hides /home/red/emacs/lisp/org/org-entities
/home/red/.emacs.d/elpa/org-20191118/ob-exp hides /home/red/emacs/lisp/org/ob-exp
/home/red/.emacs.d/elpa/org-20191118/ob-J hides /home/red/emacs/lisp/org/ob-J
/home/red/.emacs.d/elpa/org-20191118/ob-scheme hides /home/red/emacs/lisp/org/ob-scheme
/home/red/.emacs.d/elpa/org-20191118/ox-org hides /home/red/emacs/lisp/org/ox-org
/home/red/.emacs.d/elpa/org-20191118/org-datetree hides /home/red/emacs/lisp/org/org-datetree
/home/red/.emacs.d/elpa/org-20191118/org-inlinetask hides /home/red/emacs/lisp/org/org-inlinetask
/home/red/.emacs.d/elpa/org-20191118/org-attach hides /home/red/emacs/lisp/org/org-attach
/home/red/.emacs.d/elpa/org-20191118/org-ctags hides /home/red/emacs/lisp/org/org-ctags
/home/red/.emacs.d/elpa/org-20191118/ob-dot hides /home/red/emacs/lisp/org/ob-dot
/home/red/.emacs.d/elpa/org-20191118/ox-latex hides /home/red/emacs/lisp/org/ox-latex
/home/red/.emacs.d/elpa/org-20191118/org-macs hides /home/red/emacs/lisp/org/org-macs
/home/red/.emacs.d/elpa/org-20191118/ob-org hides /home/red/emacs/lisp/org/ob-org
/home/red/.emacs.d/elpa/org-20191118/ob-ledger hides /home/red/emacs/lisp/org/ob-ledger
/home/red/.emacs.d/elpa/org-20191118/ob-lua hides /home/red/emacs/lisp/org/ob-lua
/home/red/.emacs.d/elpa/org-20191118/ox hides /home/red/emacs/lisp/org/ox
/home/red/.emacs.d/elpa/org-20191118/ob-ebnf hides /home/red/emacs/lisp/org/ob-ebnf
/home/red/.emacs.d/elpa/org-20191118/ob-asymptote hides /home/red/emacs/lisp/org/ob-asymptote
/home/red/.emacs.d/elpa/org-20191118/ox-md hides /home/red/emacs/lisp/org/ox-md
/home/red/.emacs.d/elpa/org-20191118/ob-maxima hides /home/red/emacs/lisp/org/ob-maxima
/home/red/.emacs.d/elpa/org-20191118/ox-man hides /home/red/emacs/lisp/org/ox-man
/home/red/.emacs.d/elpa/org-20191118/ob-forth hides /home/red/emacs/lisp/org/ob-forth
/home/red/.emacs.d/elpa/org-20191118/ob-plantuml hides /home/red/emacs/lisp/org/ob-plantuml
/home/red/.emacs.d/elpa/org-20191118/org-timer hides /home/red/emacs/lisp/org/org-timer
/home/red/.emacs.d/elpa/org-20191118/org-bbdb hides /home/red/emacs/lisp/org/org-bbdb
/home/red/.emacs.d/elpa/org-20191118/org-colview hides /home/red/emacs/lisp/org/org-colview
/home/red/.emacs.d/elpa/org-20191118/ob-ref hides /home/red/emacs/lisp/org/ob-ref
/home/red/.emacs.d/elpa/org-20191118/ob-hledger hides /home/red/emacs/lisp/org/ob-hledger
/home/red/.emacs.d/elpa/org-20191118/ob-clojure hides /home/red/emacs/lisp/org/ob-clojure
/home/red/.emacs.d/elpa/org-20191118/ob-java hides /home/red/emacs/lisp/org/ob-java
/home/red/.emacs.d/elpa/org-20191118/org-macro hides /home/red/emacs/lisp/org/org-macro
/home/red/.emacs.d/elpa/org-20191118/org-duration hides /home/red/emacs/lisp/org/org-duration
/home/red/.emacs.d/elpa/org-20191118/org-agenda hides /home/red/emacs/lisp/org/org-agenda
/home/red/.emacs.d/elpa/org-20191118/org-archive hides /home/red/emacs/lisp/org/org-archive
/home/red/.emacs.d/elpa/org-20191118/ob-fortran hides /home/red/emacs/lisp/org/ob-fortran
/home/red/.emacs.d/elpa/org-20191118/ox-ascii hides /home/red/emacs/lisp/org/ox-ascii
/home/red/.emacs.d/elpa/org-20191118/ob-mscgen hides /home/red/emacs/lisp/org/ob-mscgen
/home/red/.emacs.d/elpa/org-20191118/org-info hides /home/red/emacs/lisp/org/org-info
/home/red/.emacs.d/elpa/org-20191118/org-faces hides /home/red/emacs/lisp/org/org-faces
/home/red/.emacs.d/elpa/org-20191118/ox-texinfo hides /home/red/emacs/lisp/org/ox-texinfo
/home/red/.emacs.d/elpa/org-20191118/org-crypt hides /home/red/emacs/lisp/org/org-crypt
/home/red/.emacs.d/elpa/org-20191118/ob-processing hides /home/red/emacs/lisp/org/ob-processing
/home/red/.emacs.d/elpa/org-20191118/ob-js hides /home/red/emacs/lisp/org/ob-js
/home/red/.emacs.d/elpa/org-20191118/ob-perl hides /home/red/emacs/lisp/org/ob-perl
/home/red/.emacs.d/elpa/org-20191118/org-eww hides /home/red/emacs/lisp/org/org-eww
/home/red/.emacs.d/elpa/org-20191118/ob-sass hides /home/red/emacs/lisp/org/ob-sass
/home/red/.emacs.d/elpa/org-20191118/org-w3m hides /home/red/emacs/lisp/org/org-w3m
/home/red/.emacs.d/elpa/org-20191118/ob-ditaa hides /home/red/emacs/lisp/org/ob-ditaa
/home/red/.emacs.d/elpa/org-20191118/ox-publish hides /home/red/emacs/lisp/org/ox-publish
/home/red/.emacs.d/elpa/org-20191118/org-pcomplete hides /home/red/emacs/lisp/org/org-pcomplete
/home/red/.emacs.d/elpa/org-20191118/ox-beamer hides /home/red/emacs/lisp/org/ox-beamer
/home/red/.emacs.d/elpa/org-20191118/org-mobile hides /home/red/emacs/lisp/org/org-mobile
/home/red/.emacs.d/elpa/org-20191118/ob-tangle hides /home/red/emacs/lisp/org/ob-tangle
/home/red/.emacs.d/elpa/org-20191118/ob-R hides /home/red/emacs/lisp/org/ob-R
/home/red/.emacs.d/elpa/org-20191118/ob-core hides /home/red/emacs/lisp/org/ob-core
/home/red/.emacs.d/elpa/org-20191118/ob-latex hides /home/red/emacs/lisp/org/ob-latex
/home/red/.emacs.d/elpa/org-20191118/ob-awk hides /home/red/emacs/lisp/org/ob-awk
/home/red/.emacs.d/elpa/org-20191118/org-element hides /home/red/emacs/lisp/org/org-element
/home/red/.emacs.d/elpa/org-20191118/org-plot hides /home/red/emacs/lisp/org/org-plot
/home/red/.emacs.d/elpa/org-20191118/ob hides /home/red/emacs/lisp/org/ob
/home/red/.emacs.d/elpa/org-20191118/org-protocol hides /home/red/emacs/lisp/org/org-protocol
/home/red/.emacs.d/elpa/org-20191118/org-version hides /home/red/emacs/lisp/org/org-version
/home/red/.emacs.d/elpa/org-20191118/org-src hides /home/red/emacs/lisp/org/org-src
/home/red/.emacs.d/elpa/org-20191118/org-docview hides /home/red/emacs/lisp/org/org-docview
/home/red/.emacs.d/elpa/org-20191118/org-list hides /home/red/emacs/lisp/org/org-list
/home/red/.emacs.d/elpa/org-20191118/ob-shen hides /home/red/emacs/lisp/org/ob-shen
/home/red/.emacs.d/elpa/org-20191118/ob-screen hides /home/red/emacs/lisp/org/ob-screen
/home/red/.emacs.d/elpa/org-20191118/ox-html hides /home/red/emacs/lisp/org/ox-html
/home/red/.emacs.d/elpa/org-20191118/ob-io hides /home/red/emacs/lisp/org/ob-io
/home/red/.emacs.d/elpa/org-20191118/ob-gnuplot hides /home/red/emacs/lisp/org/ob-gnuplot
/home/red/.emacs.d/elpa/org-20191118/ob-ocaml hides /home/red/emacs/lisp/org/ob-ocaml
/home/red/.emacs.d/elpa/org-20191118/org-id hides /home/red/emacs/lisp/org/org-id
/home/red/.emacs.d/elpa/org-20191118/ob-sed hides /home/red/emacs/lisp/org/ob-sed
/home/red/.emacs.d/elpa/org-20191118/org-loaddefs hides /home/red/emacs/lisp/org/org-loaddefs
/home/red/.emacs.d/elpa/org-20191118/org-install hides /home/red/emacs/lisp/org/org-install

121 Emacs Lisp load-path shadowings were found

**Comment by kjambunathan** [comment]
(I am on mobile phone)

Looking at the lines

home/red/.emacs.d/elpa/ox-odt-9.2.6.260/ox-odt hides

/home/red/emacs/lisp/org/ox-odt

I would have expected that the hidden 'ox-odt'  -- the second line above
-- comes from the 'org-2019whatever' elpa dir.   What were you doing with
'ox-odt' in the second path

The idea with reviewing shadow paths is to ensure that you don't have any
'stray' / 'stale'  copies lying around.  If you do find them, you can
delete the hidden files (or move those away from the load path) so that
their interference is nuked.

In my experience, THE ONLY reason LibreOffice ever complains about corrupt
file is when there are invalid XML files.  I am surprised when you say that
the XML files are valid.

Are you on Windows?  If yes, Are you using the right 'zip' and 'unzip'
programs?

If you don't mind, rename your test.odt to test.odt.zip and attach it to
issue tracker. Let me see if I can figure out what is amiss.

My gut feeling is that there is somerhing amiss in your setup: It is either
a stray ox-odt.el, a bar zip program or invalid xml files.

**Comment by revrari** [comment]
The problem is not restricted to one odt file. It occurs whenever I try to export from org, regardless of what file I am exporting. I am attaching my test.odt nonetheless. 
[test.odt.tar.gz](https://github.com/kjambunathan/org-mode-ox-odt/files/3882938/test.odt.tar.gz)

**Comment by QiangF** [comment]
I have no problem with the fix. Maybe your office version is too low.

**Comment by QiangF** [comment]
OK your attached file is corrupt with libreoffice 6.3.

**Comment by kjambunathan** [comment]
> The problem is not restricted to one odt file. It occurs whenever I try to export from org, regardless of what file I am exporting. I am attaching my test.odt nonetheless.
> [test.odt.tar.gz](https://github.com/kjambunathan/org-mode-ox-odt/files/3882938/test.odt.tar.gz)

> The problem is not restricted to one odt file. It occurs whenever I try to export from org, regardless of what file I am exporting. I am attaching my test.odt nonetheless.
> [test.odt.tar.gz](https://github.com/kjambunathan/org-mode-ox-odt/files/3882938/test.odt.tar.gz)

## Problem and Solution

The `test.odt` file you shared with me tells me that something in your init / config is forcing a trailing newline to `mimetype` file.  To verify this,

In the `*scratch*` buffer, copy the following snippet

```elisp
(let* ((mimetype "application/vnd.oasis.opendocument.text"))
    (with-temp-buffer
      (insert mimetype)
      (let ((coding-system-for-write 'utf-8))
	(write-file "mimetype"))))
```

and `C-M-x` it.  It will create `mimetype` file in the current directory.  In my case, the resulting `mimetype` file has no trailing new line.  In your case, _I am confident_ that the `mimetype` file will have a trailing newline.  If this is the case, try `eval`-ling the above snippet in a minimal Emacs session (i.e., start Emacs with `emacs -q` or `emacs -Q`).  


## Analysis


I am using LibreOffice 6.3 on Debian Linux (Unstable).  I see the same 'corrupt' message from LibreOffice.  When LibreOffice asked me to repair the file, I said "yes" and saved the repaired file as `test-repaired.odt`.

 I then used the online ODF validator at https://odfvalidator.org/.  (I am attaching the result of validation towards the end of this message).  You can compare the results for `test.odt` and `test-repaired.odt`.  Validation results of `test.odt` mentions issues with `mimetype`.  In Firefox, if you search for and highlight the _text_ of mimetype (i.e., _application/vnd.oasis.opendocument.text_) you will see that it is just the same.   But if you open the `mimetype` file in Emacs, and _notice very carefully_, you will see that `mimetype` file has a trailing newline.  Once you remove that trailing newline, and re-create a new ODT file, then LibreOffice opens the file just fine.

The key to recreating an ODT file from component files is to look at the `*Messages*` buffer, after a ODT export, and look for lines similar to 

```
Running zip -mX0 test.odt mimetype
Running zip -rmTq test.odt .
``` 

These are the steps for re-creating your ODT file _after_ stripping the trailing newline in `mimetype`.  The key command in the shell transcript below is the line `cat mimetype | tr -d '\n' > mimetype` which strips trailing newline from `mimetype`.  The re-created ODT file opens in LibreOffice wiht no issues whatsoever.

### Shell transcript for re-creating a ODT file from component files

```
~/Downloads/temp$ ls
test.odt  test.odt.tar.gz  test-repaired.odt

~/Downloads/temp$ mkdir temp1

~/Downloads/temp$ cp test.odt temp1

~/Downloads/temp$ cd temp1

~/Downloads/temp/temp1$ ls
test.odt

~/Downloads/temp/temp1$ unzip test.odt 
Archive:  test.odt
 extracting: mimetype                
   creating: META-INF/
  inflating: META-INF/manifest.xml   
  inflating: meta.xml                
  inflating: styles.xml              
  inflating: content.xml             

~/Downloads/temp/temp1$ cat mimetype 
application/vnd.oasis.opendocument.text

~/Downloads/temp/temp1$ cat mimetype | tr -d '\n' > mimetype

~/Downloads/temp/temp1$ ls
content.xml  META-INF  meta.xml  mimetype  styles.xml  test.odt

~/Downloads/temp/temp1$ rm test.odt 

~/Downloads/temp/temp1$ zip -mX0 test-new.odt mimetype
  adding: mimetype (stored 0%)

~/Downloads/temp/temp1$ zip -rmTq test-new.odt .

~/Downloads/temp/temp1$ ls
test-new.odt

~/Downloads/temp/temp1$ xdg-open test-new.odt

```


### Report from online ODF Validation at  https://odfvalidator.org/


**Result for test.odt**
```
ODF Validator Result Page
Result for test.odt

The document is NOT conformant ODF1.2!
Details:
test.odt: Info: ODF version of root document: 1.2
test.odt/mimetype: Error: The ODF package 'test.odt' contains a 'mimetype' file containing 'application/vnd.oasis.opendocument.text', which differs from the mediatype of the root document 'application/vnd.oasis.opendocument.text'!
test.odt: Error: The ODF mimetype 'application/vnd.oasis.opendocument.text' is invalid for the ODF XML Schema document!
test.odt/mimetype: Info: mimetype is not an ODFMediaTypes mimetype.
internal:/schema/odf1.2/OpenDocument-v1.2-os-manifest-schema.rng: Info: parsed.
test.odt/META-INF/manifest.xml: Info: no errors, no warnings
test.odt/mimetype: Info: 1 errors, no warnings
test.odt: Info: Media Type: application/vnd.oasis.opendocument.text
internal:/schema/odf1.2/OpenDocument-v1.2-os-schema.rng: Info: parsed.
test.odt/meta.xml: Info: no errors, no warnings
test.odt/styles.xml: Info: no errors, no warnings
test.odt/content.xml: Info: no errors, no warnings
internal:/schema/odf1.2/OpenDocument-v1.2-os-dsig-schema.rng: Info: parsed.
test.odt: Info: 2 errors, no warnings
```

**Result for test-repaired.odt**
```
ODF Validator Result Page
Result for test-repaired.odt

The document is NOT conformant ODF1.2!
Details:
test-repaired.odt: Info: ODF version of root document: 1.2
internal:/schema/odf1.2/OpenDocument-v1.2-os-manifest-schema.rng: Info: parsed.
test-repaired.odt/META-INF/manifest.xml: Info: no errors, no warnings
test-repaired.odt/mimetype: Info: no errors, no warnings
test-repaired.odt: Info: Media Type: application/vnd.oasis.opendocument.text
internal:/schema/odf1.2/OpenDocument-v1.2-os-schema.rng: Info: parsed.
test-repaired.odt/meta.xml: Info: Generator: LibreOffice/6.3.3.1$Linux_X86_64 LibreOffice_project/30$Build-1
test-repaired.odt/meta.xml: Info: no errors, no warnings
test-repaired.odt/settings.xml: Info: no errors, no warnings
test-repaired.odt/styles.xml[33,143]: Error: unexpected attribute "loext:contextual-spacing"
ing="false" fo:keep-with-next="always" ----^ test-repaired.odt/styles.xml[41,113]: Error: unexpected attribute "loext:contextual-spacing"
2cm" loext:contextual-spacing="false"/ ----^ test-repaired.odt/styles.xml[47,164]: Error: unexpected attribute "loext:contextual-spacing"
er-lines="false" text:line-number="0"/ ----^ test-repaired.odt/styles.xml[109,113]: Error: unexpected attribute "loext:contextual-spacing"
2cm" loext:contextual-spacing="false"/ ----^ test-repaired.odt/styles.xml[113,172]: Error: unexpected attribute "loext:contextual-spacing"
er" style:justify-single-word="false"/ ----^ test-repaired.odt/styles.xml[213,208]: Error: unexpected attribute "loext:contextual-spacing"
="0cm" style:auto-text-indent="false"/ ----^ test-repaired.odt/styles.xml[216,208]: Error: unexpected attribute "loext:contextual-spacing"
="0cm" style:auto-text-indent="false"/ ----^ test-repaired.odt/styles.xml[219,109]: Error: unexpected attribute "loext:contextual-spacing"
0cm" loext:contextual-spacing="false"/ ----^ test-repaired.odt/styles.xml[223,74]: Error: tag name "loext:graphic-properties" is not allowed. Possible tag names are: <map>,<paragraph-properties>,<text-properties>
<loext:graphic-properties draw:fill="none" draw:fill-color="#729fcf"/> ----^ test-repaired.odt/styles.xml[227,109]: Error: unexpected attribute "loext:contextual-spacing"
0cm" loext:contextual-spacing="false"/ ----^ test-repaired.odt/styles.xml[232,95]: Error: tag name "loext:graphic-properties" is not allowed. Possible tag names are: <map>,<paragraph-properties>,<text-properties>
l-color="#c0c0c0" draw:opacity="100%"/ ----^ test-repaired.odt/styles.xml[236,112]: Error: unexpected attribute "loext:contextual-spacing"
1cm" loext:contextual-spacing="false"/ ----^ test-repaired.odt/styles.xml[246,112]: Error: unexpected attribute "loext:contextual-spacing"
1cm" loext:contextual-spacing="false"/ ----^ test-repaired.odt/styles.xml[298,353]: Error: unexpected attribute "loext:contextual-spacing"
-number="0" style:join-border="false"/ ----^ test-repaired.odt/styles.xml[424,534]: Error: unexpected attribute "loext:rel-width-rel"
100%" loext:rel-width-rel="paragraph"/ ----^ test-repaired.odt/styles.xml[878,637]: Error: unexpected attribute "draw:fill"
29fcf" style:footnote-max-height="0cm" ----^ test-repaired.odt/styles.xml: Info: 16 errors, no warnings
test-repaired.odt/content.xml: Info: no errors, no warnings
internal:/schema/odf1.2/OpenDocument-v1.2-os-dsig-schema.rng: Info: parsed.
test-repaired.odt: Info: 16 errors, no warnings
```

The errors that mentions `loext:` can safely be ignored (in our case).  The `loext:` attributes are LibreOffice's own additions to standard ODF.  The errrors shouldn't be of any concen when opening the file with `loext:` attributes in LibreOffice itself.  IME, they pose no issues when opening with another ODF application like Google Docs etc.

**Comment by kjambunathan** [comment]
> The test.odt file you shared with me tells me that something in your init / config is forcing a trailing newline to mimetype file.

My guess is your config sets `require-final-newline`  to a non-`nil` value.

```
require-final-newline 
  User option: Whether to add a newline automatically at the end of
               the file.

```

```
require-final-newline is a variable defined in ‘files.el’.
Its value is nil

  This variable is safe as a file local variable if its value
  satisfies the predicate ‘symbolp’.
  You can customize this variable.
  Probably introduced at or before Emacs version 21.1.

Documentation:
Whether to add a newline automatically at the end of the file.

A value of t means do this only when the file is about to be saved.
A value of ‘visit’ means do this right after the file is visited.
A value of ‘visit-save’ means do it at both of those times.
Any other non-nil value means ask user whether to add a newline, when saving.
A value of nil means don’t add newlines.

Certain major modes set this locally to the value obtained
from ‘mode-require-final-newline’.

This variable is only heeded when visiting files (or saving
buffers into files they visit).  Writing data to the file system
with ‘write-region’ and the like is not influenced by this variable.

[back]
```

**Comment by kjambunathan** [comment]
> My guess is your config sets require-final-newline to a non-nil value.

> Writing data to the file system
with ‘write-region’ and the like is not influenced by this variable.

> (let* ((mimetype "application/vnd.oasis.opendocument.text"))
>     (with-temp-buffer
>       (insert mimetype)
>       (let ((coding-system-for-write 'utf-8))
> 	(write-file "mimetype"))))

I shouldn't be using `write-file` to write the mimetype file.  I should use `write-region` instead.  Anyways ... please confirm whether or my diagnosis is right.  As soon as you confirm, I will push in the needed fix.

**Comment by kjambunathan** [comment]
> The test.odt file you shared with me tells me that something in your init / config is forcing a trailing newline to mimetype file.

The fix is available in `ox-odt-9.2.6.261.tar`.  

Let me know if you continue to experience any issues with this change.

**Comment by revrari** [comment]
This fix resolved my issue. I am no longer promoted to repair a corrupt file. With much thanks.
